core/vm/logger.go

```diff
@@ -371,27 +411,30 @@ func FormatLogs(logs []StructLog) []types.StructLogRes {
 			Depth:   trace.Depth,
 			Error:   trace.ErrorString(),
 		}
-		if trace.Stack != nil {
+		if len(trace.Stack) != 0 {
 			stack := make([]string, len(trace.Stack))
 			for i, stackValue := range trace.Stack {
 				stack[i] = stackValue.Hex()
 			}
 			formatted[index].Stack = &stack
 		}
-		if trace.Memory != nil {
+		if len(trace.Memory) != 0 {
 			memory := make([]string, 0, (len(trace.Memory)+31)/32)
 			for i := 0; i+32 <= len(trace.Memory); i += 32 {
 				memory = append(memory, fmt.Sprintf("%x", trace.Memory[i:i+32]))
 			}
 			formatted[index].Memory = &memory
 		}
-		if trace.Storage != nil {
+		if len(trace.Storage) != 0 {
 			storage := make(map[string]string)
 			for i, storageValue := range trace.Storage {
 				storage[fmt.Sprintf("%x", i)] = fmt.Sprintf("%x", storageValue)
 			}
 			formatted[index].Storage = &storage
 		}
+		if trace.ExtraData != nil {
+			formatted[index].ExtraData = trace.ExtraData.CleanExtraData()
```

Comment:
I personally would prefer `CleanExtraData` -> `Seal` or `Finalize`, because `CleanExtraData` sounds like "remove ExtraData" 

But not a big problem. doesn't really matter